src/plugin.js

```diff
@@ -470,64 +493,51 @@ const vr = function(options) {
         window.requestAnimationFrame(animate);
       }
 
-      camera.getWorldDirection(cameraVector);
+      player.vr.camera.getWorldDirection(player.vr.cameraVector);
     }());
+  };
 
-    // Expose the Three.js perspective camera and rotation values on the player under the 'vr' namespace:
-    player.vr = {
-      cameraVector,
-      threeJs: {
-        scene,
-        camera,
-        renderer
-      }
-    };
-
-  }
-
-  player.on('loadstart', function() {
-    initScene();
+  player.on('loadedmetadata', function() {
+    player.vr.initScene();
   });
+  function updateVrControls(cb) {
+
+    // Add Carboard button
+    const Button = videojs.getComponent('Button');
+    const VRButton = videojs.extend(Button, {
+      constructor() {
+        Button.apply(this, arguments);
+        log(this);
+      },
+      handleClick() {
+        const event = new window.Event('vrdisplayactivate');
+
+        window.dispatchEvent(event);
+      },
+      buildCSSClass() {
+        return Button.prototype.buildCSSClass.call(this) + 'vjs-button-vr';
+      }
+    });
 
-  player.ready(function() {
-
-    function updateVrControls(cb) {
-
-      // Add Carboard button
-      const Button = videojs.getComponent('Button');
-      const VRButton = videojs.extend(Button, {
-        constructor() {
-          Button.apply(this, arguments);
-          log(this);
-        },
-        handleClick() {
-          const event = new window.Event('vrdisplayactivate');
-
-          window.dispatchEvent(event);
-        },
-        buildCSSClass() {
-          return Button.prototype.buildCSSClass.call(this) + 'vjs-button-vr';
-        }
-      });
-
-      videojs.registerComponent('VRButton', VRButton);
-      cb.addChild('VRButton', {});
+    videojs.registerComponent('VRButton', VRButton);
+    cb.addChild('VRButton', {});
 
-      // if ios remove full screen toggle
-      if (videojs.browser.IS_IOS) {
-        player.controlBar.fullscreenToggle.hide();
-      }
+    // if ios remove full screen toggle
+    if (videojs.browser.IS_IOS) {
+      player.controlBar.fullscreenToggle.hide();
     }
+  }
 
-    // mobile devices
-    if (videojs.browser.IS_ANDROID || videojs.browser.IS_IOS || videojs.browser.IS_IPAD) {
-      updateVrControls(player.controlBar);
-    }
-  });
+  // mobile devices
+  if (videojs.browser.IS_ANDROID || videojs.browser.IS_IOS || videojs.browser.IS_IPAD) {
+    updateVrControls(player.controlBar);
+  }
 
-  return {
-    changeProjection
-  };
+  return player.vr;
+};
+
+const vr = function(options) {
+  this.ready(() => initPlugin(this, options));
```

Comment:
Not waiting for player.ready caused issues on iOS devices when the source was set in the video element. Presumable because of async source setting in video.js 6